pkg/skaffold/build/tag/tag.go

```diff
@@ -32,21 +35,39 @@ type Tagger interface {
 	GenerateTag(workingDir, imageName string) (string, error)
 }
 
+const DeprecatedImageName = "_DEPRECATED_IMAGE_NAME_"
+
 // GenerateFullyQualifiedImageName resolves the fully qualified image name for an artifact.
 // The workingDir is the root directory of the artifact with respect to the Skaffold root,
 // and imageName is the base name of the image.
 func GenerateFullyQualifiedImageName(t Tagger, workingDir, imageName string) (string, error) {
+	// Supporting the use of the deprecated {{.IMAGE_NAME}} in envTemplate
+	if v, ok := t.(*envTemplateTagger); ok {
+		tag, err := v.GenerateTag(workingDir, DeprecatedImageName)
+
+		if err != nil {
+			return "", fmt.Errorf("generating deprecated envTemplate tag: %w", err)
+		}
+
+		if strings.Contains(tag, DeprecatedImageName) {
+			warnings.Printf("{{.IMAGE_NAME}} is deprecated, envTemplate's template should only specify the tag value. See https://skaffold.dev/docs/pipeline-stages/taggers/")
+
+			tag, _ = v.GenerateTag(workingDir, imageName)
+
+			return tag, nil
+		}
+
+		return fmt.Sprintf("%s:%s", imageName, tag), nil
+	}
+
 	tag, err := t.GenerateTag(workingDir, imageName)
 
```

Comment:
Could you remove this blank line?

Reply:
I'm unsure which line you are referring to.

Reply:
Line 40/64 — you didn't introduce it, but it seems wrong considering to have a blank line between the call and the error-handling.

Reply:
I'd actually suggest adding a line above 39/65.  So the code would look like:
```
}

tag, err := t.GenerateTag(workingDir, imageName)
if err != nil {
	return "", fmt.Errorf("generating tag: %w", err)
}
```